### PR TITLE
feat: Initial stab at stacks-signer design

### PR DESCRIPTION
**Pull request by netrome**
This design is not super compatible with the current version of frost-signer. Ideas for how to bring these two closer are much appreciated.

### CODE REVIEW

**Review by xoloki — APPROVED** [review]
LGTM

File: stacks-signer/README.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+```
+
+## Signer event loop
+A rough outline of the signer event loop
+
+```mermaid
+graph TD
+    A[Read configuration from file and sBTC contract] --> B{Next incoming message}
+    B -->|DKG_BEGIN| C[Generate shares]
+    C --> D[Send shares]
```
Comment (donpdonp) [inline stacks-signer/README.md:48]:
```suggestion
    C --> D[Send shares]
```

**Review by donpdonp — APPROVED** [review]

File: stacks-signer/README.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
+# Stacks signer
+The stacks signer is responsible for participating in the FROST signing scheme to
+sign sBTC peg out fulfillments from a shared wallet.
+
+Multiple signer instances communicate with each other and a coordinator using an HTTP relay.
+
+The signer interacts with th sBTC contract to read public keys for other signers
```
Comment (igorsyl) [inline stacks-signer/README.md:7]: "th sBTC"

File: stacks-signer/README.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    S[Signer]-->|Get configuration|D[sBTC contract]
+    S-->|DKG+Signing|R[Relay server]
+```
+
+## Signer design
+The signer system should contain an event loop which communicates with an HTTP relay and
+communicates with a signer entity.
+
+```rust
+pub trait NetworkedSigner {
```
Comment (igorsyl) [inline stacks-signer/README.md:21]: SignerService?

File: stacks-signer/README.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    type RelayServer;
+    type StacksNode; // Shared implementation with frost-coordinator
```
Comment (igorsyl) [inline stacks-signer/README.md:23]: ... with _stacks_-coordinator?

File: stacks-signer/README.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+    fn relay(&self) -> &Self::RelayServer;
+    fn stacks_node(&self) -> &Self::StacksNode;
+    fn signer(&self) -> &Self::Signer;
+
+    // Provided method
+    fn run(mut self);
+}
+
+pub trait Signer {
```
Comment (igorsyl) [inline stacks-signer/README.md:34]: These methods need to be aware of the `round_id`.

**Review by igorsyl — CHANGES_REQUESTED** [review]
This is great! The design of the db storing the private keys is important. Ideally, we would an append-only db to avoid risk of deleting keys.

File: stacks-signer/README.md (hunk 2 above)
Comment (netrome) [reply stacks-signer/README.md:7]: Wha? Ca't yo fil i th missin lettrs whn readin

File: stacks-signer/README.md (hunk 3 above)
Comment (netrome) [reply stacks-signer/README.md:21]: Yeah that's better!

File: stacks-signer/README.md (hunk 4 above)
Comment (netrome) [reply stacks-signer/README.md:23]: Correct!

Comment (netrome) [reply stacks-signer/README.md]: Updated.

File: stacks-signer/README.md (hunk 3 above)
Comment (netrome) [reply stacks-signer/README.md]: Updated.

File: stacks-signer/README.md (hunk 1 above)
Comment (netrome) [reply stacks-signer/README.md]: Updated :heavy_check_mark:

File: stacks-signer/README.md (hunk 5 above)
Comment (netrome) [reply stacks-signer/README.md:34]: Good point, updated.

File: stacks-signer/README.md (hunk 2 above)
Comment (xoloki) [reply stacks-signer/README.md]: It's a curious aspect of the human language ability, but it's possible in many languages to leave the vowels out and still be fully intelligible.  Some languages like Hebrew don't even have glyphs for vowels, they use diacritics which are optional.

Comment (netrome) [reply stacks-signer/README.md]: Indeed, while other languages like Japanese and Finnish are completely phonetic. And then we have English... where the grapheme to phoneme correlation is at that sweet spot where you motivated to guess how new words should be pronounced, but you're almost always wrong.

**Pull request merged**